### PR TITLE
[TACHYON-1163] ApplicationMaster should disable ContainerRequest relaxLocality when Tachyon master address is specified explicitly

### PR DESCRIPTION
**Pull request by vesense**
https://tachyon.atlassian.net/browse/TACHYON-1163

### CODE REVIEW

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 1]
```diff
@@ -178,8 +179,12 @@ public void requestContainers() throws Exception {
     String[] nodes = {mMasterAddress};
 
     // Make container request for Tachyon master to ResourceManager
+    boolean relaxLocality = true;
+    if (StringUtils.isNotEmpty(mMasterAddress) && !mMasterAddress.equals("localhost")) {
+      relaxLocality = false;
+    }
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Would `relaxLocality = mMasterAddress.isEmpty() || mMasterAddress.equals("localhost")` be sufficient? (We should add some preconditions to prevent `mMasterAddress == null` in the constructor, but that can be a separate PR)

**Comment by apc999** [comment]
+1 LGTM.

And we should cherrypick this PR into branch-0.8

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java (hunk 1 above)
Comment (vesense) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Nice. I see TACHYON-1165 has fixed that. So, I will just remove the isEmpty code.

**Comment by apc999** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
LGTM
